src/QMCHamiltonians/tests/test_coulomb_pbcAA.cpp

```diff
@@ -257,13 +267,99 @@ void test_CoulombPBCAA_3p(DynamicCoordinateKind kind)
   CHECK(caa.getValue() == Approx(-5.4954533536));
   CHECK(caa_clone.getValue() == Approx(-6.329373489));
 
-  CHECK(caa.MC0 == Approx(vmad_bcc));
-  CHECK(caa_clone.MC0 == Approx(vmad_bcc));
+  CHECK(caa.get_madelung_constant() == Approx(vmad_bcc));
+  CHECK(caa_clone.get_madelung_constant() == Approx(vmad_bcc));
 }
 
 TEST_CASE("Coulomb PBC A-A BCC 3 particles", "[hamiltonian]")
 {
   test_CoulombPBCAA_3p(DynamicCoordinateKind::DC_POS);
   test_CoulombPBCAA_3p(DynamicCoordinateKind::DC_POS_OFFLOAD);
 }
+
+
+TEST_CASE("CoulombAAListener", "[hamiltonian]")
```

Comment:
Can we fuse this test with the above test and make it testing with and without listener?

Reply:
I'm hesitant to merge further tests here.  They are already broad integration tests with unclear goals.  I'd rather see refactoring so that these are more targeted unit tests. There certainly is room to factor out some shared setup code.  I'll look at doing that.

Reply:
I think both mw_evaluate functions w/ w/o listeners should be available to use at the same time. That is why I think we should not avoid putting it as a fully separate unit test.

Reply:
I agree but I'm used to crudely think of unit tests being roughly per function.  But I see your point here, if we want to skip reporting we need to be able to call both with the same setup.  That should work fine let me add a call to mw_evaluate(). I'm not sure that justifies dropping one of the other tests though.